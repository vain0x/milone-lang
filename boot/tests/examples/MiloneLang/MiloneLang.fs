module rec MiloneLang.Program

open MiloneLang.Polyfills
open MiloneLang.Lexing
open MiloneLang.Types
open MiloneLang.Self

let fileReadAllText (fileName: string): string =
  (__nativeFun "file_read_all_text" 1) fileName

[<EntryPoint>]
let main _ =
  doSelf fileReadAllText
