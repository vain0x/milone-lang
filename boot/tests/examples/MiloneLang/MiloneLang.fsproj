<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>netcoreapp2.1</TargetFramework>
  </PropertyGroup>
  <ItemGroup>
    <Compile Include="Compatible.fs" />
    <Compile Include="Types.fs" />
    <Compile Include="Records.fs" />
    <Compile Include="Helpers.fs" />
    <Compile Include="Lexing.fs" />
    <Compile Include="Parsing.fs" />
    <Compile Include="AstToHir.fs" />
    <Compile Include="Bundling.fs" />
    <Compile Include="NameRes.fs" />
    <Compile Include="Typing.fs" />
    <Compile Include="MainHoist.fs" />
    <Compile Include="ClosureConversion.fs" />
    <Compile Include="EtaExpansion.fs" />
    <Compile Include="Hoist.fs" />
    <Compile Include="Monomorphizing.fs" />
    <Compile Include="Mir.fs" />
    <!-- <Compile Include="CIrGen.fs" /> -->
    <!-- <Compile Include="CPrinting.fs" /> -->
    <Compile Include="Self.fs" />
    <Compile Include="MiloneLang.fs" />
  </ItemGroup>
</Project>