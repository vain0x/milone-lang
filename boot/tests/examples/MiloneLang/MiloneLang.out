Load MiloneLang
Load Polyfills
Load Lexing
Load Types
Load Helpers
Load Records
Load Self
Load Parsing
Load AstToHir
Load Bundling
Load NameRes
Load Typing
Load MainHoist
Load ClosureConversion
Name resolution
Type inference
Hoist main
Closure conversion
$? = 0
